go/api/response_handlers.go

```diff
@@ -280,6 +280,25 @@ func handleBooleanResponse(response *C.struct_CommandResponse) (Result[bool], er
 	return CreateBoolResult(bool(response.bool_value)), nil
 }
 
+func handleBooleanArrayResponse(response *C.struct_CommandResponse) ([]Result[bool], error) {
+	defer C.free_command_response(response)
+
+	typeErr := checkResponseType(response, C.Array, false)
+	if typeErr != nil {
+		return nil, typeErr
```

Comment:
Same here, Result has functions in response_types.go to use in return statements.
In this case CreateNilBoolResult() should be used.
```suggestion
		return CreateNilBoolResult(), typeErr
```

Reply:
Same explaination still apply here, it's a slice (array) e.g., []Result[bool] , not Result[bool]